### PR TITLE
generated TabIds and encrypted postMessage messaging

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
This PR uses either broadcast messages sent with `chrome.runtime` or encrypted `postMessage`
- wired `ajax_progress` callback messages directly to individual `pgp_block` frames when needed
- encrypted window messages (encryption key isn't rotated -- it stays in the local storage forever)

issue #5329 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/browser/browser-msg.ts [hunk 1]
```diff
@@ -430,31 +371,56 @@
     });
   };
 
-  private static sendToParentWindow = (name: string, bm: Dict<unknown> & { messageSender: Bm.Dest }, responseName?: string) => {
-    const raw: Bm.RawWindowMessageWithSender = {
+  protected static listenForWindowMessages = (dest: Bm.Dest[]) => {
+    window.addEventListener('message', async e => {
```
[the remark this answers is not in the document]
Comment (sosnovsky) [reply extension/js/common/browser/browser-msg.ts:375]: What about this warning, probably we can add check if message origin is `mail.google.com` or extension, so we'll process only messages from them?

**Review by sosnovsky — APPROVED** [review]
Works great, didn't found any issues, well done 👍

**Comment by tomholub** [comment]
I think I'll do a release just before this is merged

**Review by tomholub — COMMENTED** [review]
I like this PR 👍

**Review by sosnovsky — APPROVED** [review]

**Pull request merged**